### ISSUE TITLE
eSpeak doesn't read the rest of message after hyperlinked text in Persian language

### ISSUE THREAD

**Issue by rbakhshi72** [espeak-ng/espeak-ng#1071]
Hello, 
As you know, if a well implemented user interface has used in an app, for reading the text containing a hyperlink, TalkBack produces an auditory icon as the hyperlinked text is spoken and pauses before and after hyperlinked text. Likewise, the linked text appears in the "Links" sub-menu of TalkBack menu.

If you use eSpeak NG as your preferred engine, Talkback sometimes doesn’t continue to read the rest of text after pausing for hyperlinked text. This issue just exists in Persian language settings. In fact, there is no this issue for other languages.

I will write the steps of producing this issue in detail.

Prerequisite: Android 9, TalkBack 12.1, eSpeak NG 2.6.1 and Telegram 8.4.4.

Steps to reproduce
1. Open TalkBack menu and click on Text-to-speech settings.
2. Choose Persian for your language. See “1.jpg”.

Notice: A. You don’t need to change your device language.
B. Persian is dual language. It means you are able to read both English and Persian texts simultaneously.
C. We are considering this issue on Voice variant, Male (Default). See “2.jpg”.

3. Open “3.txt” from folder (bug) and copy the whole text.
4. Open Telegram and paste it in a chat.
5. You will see that TalkBack doesn’t read the rest of the text after hashtag. Hashtag is a hyperlink  here. It just says, “hash khabar”. Watch 3-1.mp4 from the folder.
6. Now open “4.txt” and copy the whole text.
7. Open Telegram and paste it in a chat.
8. You will see that TalkBack read the rest of the text after hashtag right now. Watch 4-1.mp4.

Why does TalkBack read the rest message after hashtag in “4.txt”?
I will tell you why.

I wrote above that TalkBack pauses half an second between hyperlinked text and regular text. Likewise, You know that TalkBack pauses when it reaches a punctuation like comma or period.
If you open “3.txt” and read the first line character to character, you will see that there is a colon (:) after ر, so, TalkBack should pauses twice here, one time for hyperlinked text and one time for colon (:).
In “4.txt”, I removed colon (:) after ر, so TalkBack is able to read the rest of the message completely after hashtag without any problem.

Notice that we don’t have this issue on voice variant, Male (Variant 2), voice variant, Female (Default), voice variant, Female (Variant 2), voice variant, Female (Variant 3), voice variant, Female (Variant 4), voice variant, Female (Variant 5), voice variant, Young (Male) and voice variant, Young (Female).

 
Now let’s consider this issue for English texts.

As before I wrote, Persian is dual language. It means you are able to read both English and Persian texts simultaneously. TalkBack sometimes doesn’t read the rest of message after hyperlinked text in English, too.

Steps to reproduce
1. Open TalkBack menu and click on Text-to-speech settings.
2. Choose Persian for your language. See “1.jpg”.
3. Open “5.txt” and copy the whole text.
4. Open Telegram and paste it in a chat.
5. You will see that TalkBack doesn’t read the rest of the text after hashtag. Hashtag is a hyperlink  here. It just says, “hash bug, hash accessibility”. Watch 5-1.mp4.
6. Now open “6.txt” and copy the whole text.
7. Open Telegram and paste it in a chat.
8. You will see that TalkBack read the rest of the text after hashtags right now. Watch 6-1.mp4.
Why does TalkBack read the rest message after Hashtag in “6.txt”?
I will tell you why.

I wrote above that TalkBack pauses half an second between hyperlinked text and regular text. Likewise, you know that TalkBack pauses when it reaches an empty line.
If you open “5.txt” and read the first line character to character, you will see that there is an empty line after “accessibility”, so TalkBack should pauses twice here, one time for hyperlinked text and one time for empty line.
In “6.txt”, I removed the empty line after “accessibility”, so TalkBack reads the rest of the message completely after hashtags without any problem.

Notice: we don’t have this issue when we choose English for TTS language. See “7.jpg” and watch “8.mp4”

As a result, When Talkback pauses twice after hyperlinked text immediately, It can not read the rest of the text. We have this issue when we choose Persian for TTs Language.
 
I   @##hope this information help you to fix this issue as soon as possible.

[Bug.zip](https://github.com/espeak-ng/espeak-ng/files/8009811/Bug.zip)
#890 

- [ ] 

1. [ ] 

#- [ ] @

**Comment by alex19EP** [comment]
it is a specific to android problem. I am trying to fix that because it affects not only Persian but at least Russian too.
